src/app/pcasts/controllers/listening_history_controller.py

```diff
@@ -0,0 +1,31 @@
+import json
+from . import *
+
+class ListeningHistoryController(AppDevController):
+
+  def get_path(self):
+    return '/history/listening/'
+
+  def get_methods(self):
+    return ['GET', 'POST']
+
+  @authorize
+  def content(self, **kwargs):
+    if request.method == 'GET':
+      user = kwargs.get('user')
+      offset = int(request.args['offset'])
+      max_hs = int(request.args['max'])
+      histories = \
+        listening_histories_dao.get_listening_history(user, max_hs, offset)
+
+    elif request.method == 'POST':
+      user = kwargs.get('user')
+      episode_info = {long(k): v for k, v in json.loads(request.data).items()}
+      histories = listening_histories_dao \
+        .create_or_update_listening_histories(episode_info, user)
+      app.logger.info(
```

Comment:
Should the logger be pushing user_id information? Isn't that dangerous from a security standpoint?

Reply:
maybe not, but every endpoint that creates or deletes something does this, so that could be changed in a future PR

Reply:
the sessions should prevent any malicious api calls, and I'm not too knowledgeable about how secure logs on a server are, but i guess it doesnt help. I just thought it would be convenient for debugging purposes to know the id (if we wanna make some db requests), but we can definitely do without it if its dangerous

Reply:
Production PRs should not include doing logs of DB specific ids.

Reply:
so what do you think we should log, if anything?

Reply:
also, I'm curious why logging id's isn't best practice? I don't understand how the logs would be accessed by malicious users in the first place, and even if they could, what could they do with it, since identity is confirmed through a session?

Reply:
Because malicious users who have access to server logs should not have awareness of DB specific information. For example: printing in logs credit card information after successful submission. You wouldnt log that, as such you shouldn’t print in production similarly. For debugging purposes, logs are useful but shouldn’t be leveraged in production instances for the sake of security guarantees of our service

Reply:
Yeah credit card info definitely wouldnt make much sense b/c with it alone, you can do some malicious stuff, but I guess outside of the context of our app, user id doesnt mean as much, I cant think of what they could do with it alone. And I thought the entire point of logging was primarily for production instances, so that if there is some bug we can trace down the calls that lead up to it, given user who reports it and timestamp. Not sure how useful they would be just in such a controlled test environment, if that makes sense. At least from what I saw from my (limited) experience in industry so far, we used logs heavily in production instances to track down bugs reported by users

Reply:
What I am trying to say is that logging of DB specific stuff. like DB internal IDs, env variables, etc. is security flaw. Things that people can reverse engineer with malicious intent. If you feel as though the user id serves no risk, than it is okay to include in the logs. The problem is when you start logging information that sys admins shouldn't necessarily have access to in the case that a sys admin becomes malicious

Reply:
Ok that's a fair point, at the current state, user id doesn't maintain any risk with the endpoints we have currently, but I guess there is no guarantee for that in the future, and it is redundant with the username being logged as well. I'll make another PR to go through and remove all the user_ids, and I also plan on logging in json format instead of raw string, should be easier to query for stuff that way